### PR TITLE
ML-2296: Add support for datastore profiles

### PR DESCRIPTION
**Pull request by alxtkr77**
[ML-2296](https://jira.iguazeng.com/browse/ML-2296)

### CODE REVIEW

File: storey/utils.py [hunk 1]
```diff
@@ -138,6 +138,12 @@ def get_remaining_path(url):
 
 def url_to_file_system(url, storage_options):
     scheme, remaining_path = get_remaining_path(url)
+    if url.startswith("ds://"):
+        parsed_url = urlparse(url)
+        if parsed_url.password:
+            scheme = parsed_url.password
```
Comment (gtopper) [inline storey/utils.py:144]: Can we add a test to show what this looks like?

File: storey/utils.py [hunk 2]
```diff
@@ -138,6 +138,12 @@ def get_remaining_path(url):
...
+        else:
+            raise ValueError("Datastore profile URL expecting to have underlying scheme embedded as a password")
```
Comment (gtopper) [inline storey/utils.py]:
```suggestion
            raise ValueError("Datastore profile URL is expected to have underlying scheme embedded as password")
```

File: storey/utils.py (hunk 1 above)
Comment (alxtkr77) [reply storey/utils.py:144]: done

File: integration/test_filesystems_integration.py [hunk 3]
```diff
@@ -672,3 +673,10 @@ def test_get_path_utils():
     schema, path = get_remaining_path(url)
     assert path == "mycontainer/path/to/object.csv"
     assert schema == "wasbs"
+
+
+def test_ds_get_path_utils():
```
Comment (gtopper) [inline integration/test_filesystems_integration.py]: I think this is basically a unit test, so better move it there.

Comment (alxtkr77) [reply integration/test_filesystems_integration.py]: Hmm, you probably right , and test_get_path_utils() should be moved as well. But there is no unit tests for the utils.py module. Where would you put one?

Comment (gtopper) [reply integration/test_filesystems_integration.py]: Feel free to create a `test_utils.py` file under `tests/` and move anything that's appropriate to move in there.

**Review by gtopper — APPROVED** [review]
LGTM

**Pull request merged**